mono/mini/aot-compiler.c

```diff
@@ -4333,7 +4333,7 @@ static void
 add_generic_class (MonoAotCompile *acfg, MonoClass *klass, gboolean force, const char *ref)
 {
 	/* This might lead to a huge code blowup so only do it if neccesary */
-	if (!mono_aot_mode_is_full (&acfg->aot_opts) && !force)
+	if (!acfg->aot_opts.llvm_only && !mono_aot_mode_is_full (&acfg->aot_opts) && !force)
 		return;
```

Comment:
llvm_only is supposed to be a superset of full_aot, so is_full () should probably return true when llvm_only is enabled.